torch/distributed/distributed_c10d.py

```diff
@@ -753,6 +778,82 @@ def recv(tensor,
         return src
 
 
+@contextlib.contextmanager
+def _batch_p2p_manager(backend):
+    if backend == Backend.NCCL:
+        ProcessGroupNCCL._group_start()
+    try:
+        yield
+    finally:
+        if backend == Backend.NCCL:
+            ProcessGroupNCCL._group_end()
+
+
+def batch_isend_irecv(op_list,
```

Comment:
We should clarify the behavior of this API for Gloo. As per my understanding we should probably mention in the docs that this API works for CPU tensors on Gloo and as a result we should also add tests validating this works correctly for Gloo.